packages/modules/imodel-browser/src/containers/iModelGrid/IModelGrid.tsx

```diff
@@ -109,7 +109,12 @@ export const IModelGrid = ({
   viewMode,
   maxCount,
 }: IModelGridProps) => {
-  const [sort, setSort] = useState<IModelSortOptions | undefined>(sortOptions);
+  const [sort, setSort] = useState<IModelSortOptions>(sortOptions);
+
+  useEffect(() => {
```

Comment:
do you need this useEffect? when the props change (sortOptions) wouldnt it remount the component and set the right state?

Reply:
When the component remounts it wouldn't update the state. The sortOptions we're setting in the useState is only applied on the first initialization of the component, afterwards we have to use the setSort function to update the state.
I can push a change to this PR  to remove that state altogether since we can use directly the value coming from the props and we don't need to put it in the state. 
I initially kept it cuz we were using setSort in the Table component
```
onSort={() =>
  setSort({ sortType: "name", descending: !sort.descending })
}
```
but after some testing it seems like this onSort props isn't doing anything and it would work without it just fine.
![image](https://github.com/user-attachments/assets/f178e2e5-9d94-47a7-899f-39711ec90865)

Reply:
The sort is actually handled server side by the callback, removing that state basically asks the server to sort one way and the table to sort a different way, if the data is not all imported, the table will sort only part of the data that is received. (So, if you do not have alll the data, and revert the sort, you will not get more data, you will simply sort what you fetched so far, which will give wrong results.)

Reply:
I tested it with a long list of iModels I have (that requires scrolling to the bottom to fetch all the data) and changing the sort method still displays the proper results (i.e. the results at the bottom of the full list is what is being displayed at the top)

Reply:
It will if you have already fetched the data (scrolled to the end) then sort, but I'm really surprised if you refresh the page, only have some of the data and change sorting throught the table column header that you will actually get all the data, since locallly you wouldnt have this data. (It is different if the sort picker is outside the table, in which case it would cause the full refresh anyway because the outside prop have changed and get you the proper fetch.)

Reply:
Here's a gif where I removed the onBottomReached function, so the list doesn't populate the whole list of iModels. You can see that after I change the sort to descending, we're still getting the values at the bottom of the full list and not just the truncated data.
![sort-comment](https://github.com/user-attachments/assets/6dbddf43-6188-47c7-9896-e87948f3a02f)

Reply:
Well, as I mention, yes this works because you are changing the props. If you use the sorting of the table (clicking on the header) that would likely not work.

Reply:
ok I updated the PR to fix sorting from the table column as well